src/main/schema.ts

```diff
@@ -1,29 +1,19 @@
 export const toggleSchema: object = {
     type: 'object',
-    properties: {
-        toggles: {
-            type: 'array',
-            items: {
-                type: 'object',
-                properties: {
-                    id: {
-                        type: 'string',
-                    },
-                    description: {
-                        type: 'string',
-                    },
-                    fraction: {
-                        type: 'number',
-                        minimum: 0.0,
-                        maximum: 1.0,
-                    },
-                    comment: {
-                        type: 'string',
-                    },
+    patternProperties: {
+        '^[A-Za-z0-9 -_.]': {
```

Comment:
Commented on the readme but really it applies here: This is only testing the first character; if we want to constrain the entire property name to this character set, need `*$` at the end